operators/blackbox_docker_operator.py

```diff
@@ -35,3 +35,15 @@ def __init__(self, *args, **kwargs):
             raise ValueError('Must set DECRYPTED_SETTINGS and DESTINATION_SETTINGS environment variables')
 
         self.command = '/bin/bash -ce "airflow_scripts/concat_settings.sh; {}"'.format(self.command)
+
+    def _run_image(self):
+        '''
+        Remove containers, even if the command fails.
+        '''
+        try:
+            return super()._run_image()
+
+        except AirflowException as e:
+            self.cli.remove_container(self.container['Id'])
```

Comment:
The upstream `_run_image()` method only runs this code if `self.auto_remove` is set:

https://github.com/apache/airflow/blob/aeea71274d4527ff2351102e94aa38bda6099e7f/airflow/providers/docker/operators/docker.py#L269-L270

Probably not particularly important for our use-case, but maybe something we should set too to make sure that `auto_remove` behaves as expected?

Reply:
Good thought – done!